### DISCUSSION TITLE
Plugin for regex replacements during import

### DISCUSSION THREAD

**Discussion by edgars-supe** [beetbox/beets#4047] [Show and tell]
Hello!

When I first discovered beets a few months ago, I decided not to use it, because MusicBrainz uses non-ASCII apostrophes, quotation marks and hyphens, and I didn't have a way to make it not do that. Recently, faced with importing a whole bunch of music into my library, I thought to give beets another go. And wrote a plugin that can replace anything for the fields you want.

Introducing [ImportReplace](https://github.com/edgars-supe/beets-importreplace)! It will run all the replacements you give it on all the fields you specify. The configuration in the example replaces the aforementioned Unicode chars with their ASCII counterparts. If you want, you can replace all `a` characters with `x`. I tried it, it's pretty amusing. Anyway, this plugin could be quite useful for #2558.

It's my first (well, second, ish) contribution to beets, and I've never used Python before, so constructive criticism and suggestions are very much welcome! (even about the `setup.py` file, really - anything and everything)

**Comment by sampsyo** [comment]
Wow; just wanted to say that this looks very snazzy! Nice work!

**Reply by edgars-supe** [reply]
Thanks! :)
Should I add it to [this list](https://beets.readthedocs.io/en/stable/plugins/index.html#other-plugins)?

**Reply by sampsyo** [reply]
That would be great; yes!
